src/libraries/Pool.sol

```diff
@@ -365,7 +368,8 @@ library Pool {
                 ) ? params.sqrtPriceLimitX96 : step.sqrtPriceNextX96,
                 state.liquidity,
                 state.amountSpecifiedRemaining,
-                swapFee
+                // use the effective fee
+                uint24(cache.protocolFee + swapFee - (uint32(cache.protocolFee) * uint32(swapFee)) / MAX_EFFECTIVE_FEE)
```

Comment:
i wonder if this should be like
`cache.protocolFee.calculateEffectiveFee(swapFee)`
and put it in the ProtocolFeeLibrary?

and then we can get rid of `MAX_EFFECTIVE_FEE` and just use `PIPS_DENOMINATOR` instead?

Although it feels weird to have to choose between ProtocolFeeLibrary and SwapFeeLibrary here hahah

Reply:
hmm yeah that feels weird putting it in the protocol fee library because it's not just about protocol fees